File: c/fizzbuzz.c
#include <stdio.h>
#include <stdlib.h>

char *itoa(const int num) 
{
    char *out = malloc((sizeof(char) * abs(num / 10))+1);
    sprintf(out, "%d", num);
    return out;
}

char *fizzbuzzer(int num) 
{
    if (num % 15 == 0) 
    {
        return "FizzBuzz";
    }
    else if (num % 3 == 0) 
    {
        return "Fizz";
    } 
    else if (num % 5 == 0) 
    {
        return "Buzz";
    } 
    else 
    {
        return itoa(num);        
    } 
}

int main(int argc, char *argv[]) 
{
    for (int num = 1; num <= 100; num++) 
    {
        printf("%s\n", fizzbuzzer(num));
    }
}

File: java/fizzbuzz.java
import java.util.Map;

class FizzBuzz {

    private final Map<Integer, String> replacementMap;

    public FizzBuzz() {
        replacementMap = Map.of(
                3, "Fizz",
                5, "Buzz"
        );
    }

    public final String fizzbuzzer(int num) {
        StringBuilder out = new StringBuilder();

        for (Map.Entry<Integer, String> entry : replacementMap.entrySet()) {
            if (num % entry.getKey() == 0) {
                out.append(entry.getValue());
            }
        }

        if (out.length() == 0) {
            out.append(num);
        } 

        return out.toString();
    }

    public static void main(String[] args) {
        FizzBuzz fb = new FizzBuzz();

        for (int num = 1; num <= 100; num++) 
        {
            System.out.println(fb.fizzbuzzer(num));
        }
    }
}